src/ncp/ncp_uart.cpp

```diff
@@ -270,12 +270,101 @@ void NcpUart::HandleError(ThreadError aError, uint8_t *aBuf, uint16_t aBufLength
 #ifdef __cplusplus
 extern "C" {
 #endif
-void otCliLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, va_list aAp)
+
+#define LOG_APPEND(CONST_STR) \
+    strcpy(logString + offset, CONST_STR); \
+    offset += sizeof(CONST_STR) - 1; \
+
+void otCliLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, va_list args)
 {
-    (void)aLogLevel;
-    (void)aLogRegion;
-    (void)aFormat;
-    (void)aAp;
+    char logString[128];
+    unsigned int offset = 0;
+
+    switch (aLogLevel)
+    {
+    case kLogLevelNone:
+        LOG_APPEND("NONE ");
+        break;
+
+    case kLogLevelCrit:
+        LOG_APPEND("CRIT ");
+        break;
+
+    case kLogLevelWarn:
+        LOG_APPEND("WARN ");
+        break;
+
+    case kLogLevelInfo:
+        LOG_APPEND("INFO ");
+        break;
+
+    case kLogLevelDebg:
+        LOG_APPEND("DEBG ");
+        break;
+
+    default:
+        LOG_APPEND("???? ");
+        break;
+    }
+
+    switch (aLogRegion)
```

Comment:
Commit #1046 added support for this through a config option. I suggest to use that instead.